web/modules/custom/rdf_entity_provenance/rdf_entity_provenance.behat.inc

```diff
@@ -0,0 +1,306 @@
+      foreach ($entity as $key => $value) {
+        if (array_key_exists($key, $aliases)) {
+          $values[$aliases[$key]] = $value;
+        }
+        else {
+          throw new \Exception("Unknown column '$key' in the provenance data table.");
+        }
+      };
+
+      if (empty($values['provenance_entity'])) {
+        throw new \Exception('The provenance entity is required.');
+      }
+
+      // Convert user friendly values to machine values.
+      $values = $this->convertValueAliases($values);
+
+      // Provide default values.
+      $values = $this->provideDefaultValues($values);
+
+      $this->createActivity($values);
+    }
+  }
+
+  /**
+   * Checks the number of available provenance activities.
+   *
+   * @param int $number
+   *   The expected number of activities.
+   *
+   * @throws \Exception
+   *   Throws an exception when the expected number is not equal to the given.
+   *
+   * @Then I should have :number provenance activity
+   * @Then I should have :number provenance activities
+   */
+  public function assertActivitiesCount(int $number): void {
+    $actual = \Drupal::entityQuery('rdf_entity')
+      ->condition('rid', 'provenance_activity')
+      ->count()
+      ->execute();
+    if ($actual != $number) {
+      throw new \Exception("Wrong number of activities. Expected number: $number, actual number: $actual.");
+    }
+  }
+
+  /**
+   * Asserts that an rdf entity has a provenance activity related.
+   *
+   * @param string $entity_label
+   *   The entity label that will be checked that has a provenance activity
+   *   related.
+   * @param string $entity_type
+   *   The entity bundle of the entity that will be checked for having a
+   *   provenance activity.
+   *
+   * @throws \Exception
+   *   Thrown when an entity does not have a provenance activity record related.
+   *
+   * @Then the :entity_label :entity_type should have a provenance activity related
+   */
+  public function assertActivityRelatedToEntity(string $entity_label, string $entity_type): void {
+    $entity_type = str_replace(' ', '_', $entity_type);
+    $rdf_entity = $this->getRdfEntityByLabel($entity_label, $entity_type);
+    /** @var \Drupal\rdf_entity_provenance\ProvenanceHelperInterface $provenance_helper */
+    $provenance_helper = \Drupal::service('rdf_entity_provenance.provenance_helper');
+    $activity = $provenance_helper->getProvenanceByReferredEntity($rdf_entity->id());
+    if (empty($activity)) {
+      throw new \Exception("The entity labeled {$rdf_entity->label()} does not have a provenance activity related.");
+    }
+  }
+
+  /**
+   * Asserts that a list of fields are disabled.
+   *
+   * @param string $fields
+   *   A list of comma separated field labels.
+   *
+   * @throws \Exception
+   *   Thrown when a field is not found by name or is not disabled.
+   *
+   * @Then the following fields should be disabled :fields
+   */
+  public function givenDisabledFields(string $fields): void {
```

Comment:
Put it in joinup_core? I cannot put it in a Trait since it is a step definition. We only need it in rdf_entity_provenance up to now. I can move it in joinup_core or joinup.